src/webapp/reports/autogenerated-forms/SectionsTabs.tsx

```diff
@@ -117,23 +128,104 @@ const TabPanel: React.FC<TabProps> = React.memo(props => {
+            direction={direction}
+            showLeftFade={showLeftFade}
+            showRightFade={showRightFade}
+            onClick={() => handleScroll(direction)}
+        >
+            {direction === "right" && <ChevronRight className={classes.scrollIcon} />}
+            {direction === "left" && <ChevronLeft className={classes.scrollIcon} />}
+        </StyledIconButton>
+    );
+});
+
 const SectionsTabs: React.FC<TabPanelProps> = React.memo(props => {
     const { sections, dataFormInfo } = props;
-    const [activeTab, setActiveTab] = React.useState(0);
-    const handleChange = (event: React.ChangeEvent<{}>, value: number) => {
+
+    const [activeTab, setActiveTab] = useState(0);
+    const [showLeftFade, setShowLeftFade] = useState(false);
+    const [showRightFade, setShowRightFade] = useState(true);
+    const tabsRef = useRef<HTMLButtonElement>(null);
+
+    const handleChange = (_event: React.ChangeEvent<{}>, value: number) => {
         setActiveTab(value);
     };
 
+    const tabContainer = document.querySelector(".MuiTabs-scroller.MuiTabs-scrollable");
+    const handleScroll = useCallback(
+        (direction: TabScrollButtonDirection) => {
+            const scrollOffset = direction === "left" ? -500 : direction === "right" ? 500 : undefined;
+
+            return tabContainer?.scrollBy({
+                left: scrollOffset,
+                behavior: "smooth",
+            });
+        },
+        [tabContainer]
+    );
+
+    useEffect(() => {
+        const scrollButtons = document.querySelectorAll(".MuiTabs-scrollButtons");
+        if (!tabContainer) return;
+
+        const updateFadeVisibility = () => {
+            const { scrollLeft, scrollWidth, clientWidth } = tabContainer;
+            const scrollRight = scrollWidth - clientWidth - scrollLeft;
+
+            setShowLeftFade(scrollLeft > 0);
+            setShowRightFade(scrollRight > 10);
+        };
+
+        tabContainer.addEventListener("scroll", updateFadeVisibility);
+        updateFadeVisibility();
+
+        const handleScrollButtonClick = () => {
+            setTimeout(updateFadeVisibility, 100);
+        };
+
+        scrollButtons.forEach(button => {
+            button.addEventListener("click", handleScrollButtonClick);
+        });
+
+        return () => {
+            tabContainer.removeEventListener("scroll", updateFadeVisibility);
+            scrollButtons.forEach(button => {
+                button.removeEventListener("click", handleScrollButtonClick);
+            });
+        };
+    }, [tabContainer]);
+
     return (
         <Box sx={{ width: "100%" }}>
             <StyledAppBar position="sticky" color="default">
-                <Tabs
+                <ScrollButton
+                    direction="left"
+                    showLeftFade={showLeftFade}
+                    showRightFade={showRightFade}
+                    handleScroll={handleScroll}
+                />
+                <ScrollButton
```

Comment:
The tabs component already include a prop to create custom buttons for the arrows: `ScrollButtonComponent`

https://v4.mui.com/api/tabs/#props
https://v4.mui.com/api/tab-scroll-button/#props
![image](https://github.com/user-attachments/assets/c3a8ed23-9023-4d8e-b389-538bedd5404a)
 
Are we creating the buttons from scratch because the prop was not flexible enough? If yes that's ok your implementation is correct.

Reply:
Yes, the prop is not flexible enough. The left scroll button will always disappear when the user is scrolled to the left and one of the requirements of this task is to always show both scroll buttons.